test/nightly/crud.test.ts

```diff
@@ -0,0 +1,127 @@
+            locations = await LocationListStep.getLocations(context);
+        }
+
+        rgName = randomUtils.getRandomHexString(12);
+    });
+
+    test('Create Resource Group (single)', async () => {
+        await runWithTestActionContext('createResourceGroup', async context => {
+            const testInputs: (string | RegExp)[] = [rgName, locations[0].displayName!];
+            await context.ui.runWithInputs(testInputs, async () => {
+                await createResourceGroup(context, testSubscription);
+            });
+
+            assert.ok(await resourceGroupExists(context, rgName));
+        });
+    });
+
+    test('Create Resource Groups (all locations)', async () => {
+        await Promise.all(locations.map(async l => {
+            await runWithTestActionContext('createResourceGroup', async context => {
+                const testInputs: (string | RegExp)[] = [`${rgName}-${l.name}`, l.displayName!];
+                await context.ui.runWithInputs(testInputs, async () => {
+                    await createResourceGroup(context, testSubscription);
+                });
+
+                assert.ok(await resourceGroupExists(context, `${rgName}-${l.name}`));
+            });
+        }));
+    });
+
+    test('Get Resources', async () => {
+        const subscriptionTreeItems = await ext.appResourceTree.getChildren();
+        assert.ok(subscriptionTreeItems.length > 0);
+        for (const subscription of subscriptionTreeItems) {
+            const groupTreeItems = await ext.appResourceTree.getChildren(subscription as AzExtParentTreeItem);
+            await Promise.all(groupTreeItems.map(async g => {
+                const children = await ext.appResourceTree.getChildren(g as AzExtParentTreeItem);
+                console.log(children);
+            }));
+        }
+
+        assert.ok(true);
+    });
+
+    test('Delete Resource (EnterName) - Fails when invalid', async () => {
+        await settingUtils.updateGlobalSetting('deleteConfirmation', 'EnterName');
+        await runWithTestActionContext('Delete Resource', async context => {
+            await context.ui.runWithInputs([rgName, 'rgName'], async () => {
+                try {
+                    await deleteResourceGroupV2(context);
+                } catch (_) {
+                    console.debug('Expected error: ', _);
+                    // expected to fail here
+                }
+                assert.ok(await resourceGroupExists(context, rgName));
+            });
+        });
+    });
+
+    test('Delete Resource (EnterName)', async () => {
+        await settingUtils.updateGlobalSetting('deleteConfirmation', 'EnterName');
+        await runWithTestActionContext('Delete Resource', async context => {
+            await context.ui.runWithInputs([rgName, rgName], async () => {
+                await deleteResourceGroupV2(context);
+            });
+
+            // ext.azureResourceProvider.onDidChangeResource!(async e => {
```

Comment:
Because the `deleteResourceGroupV2` command just starts the delete and doesn't actually await it, we don't get any confirmation that the delete happened. That makes it difficult for me to check to see if it actually deleted.

It emits a tree state change, so does anyone know how to watch for that event?